### ISSUE TITLE
Allow configuration of package dependencies and schema conversions

### ISSUE THREAD

**Issue by ahl** [oxidecomputer/typify#58]
Currently, typify can generate code that depends on `uuid` and `chrono`. It might be useful to enable support for other packages or it might be useful to deny use, say, of `uuid` and represent a `{ "type": "string", "format": "uuid" }` as a `String` rather than as an `uuid::Uuid`. One could even imagine allowing substitutes for the `std` types for arrays, maps, and sets.

In addition, it could be useful to allow consumers to specify new associations between formats and Rust types or to override existing ones. For example, a user might want to use `rust_decimal::Decimal" for the "decimal" format or `chrono::naive::NaiveDate` for the "date" format (rather than `chrono::Date`).

We could modify the macro, builder, and command-line interfaces to allow this (perhaps all through a shared `Settings` type). e.g.

```rust
import_types!(
    schema = "../example.json",
    allow_packages = [uuid, chrono],
    additional_string_formats = {
        "decimal" = "rust_decimal::Decimal",
        "date" = "chrono::naive::naiveDate",
    },
);
```

**Comment by ahl** [comment]
I've evolved my thinking on this a bit to go beyond formats: I'd like to be able to specify schemas or schema patterns and let the consumer define the replacement. For example:

```
{
  "type": "string",
  "format": "uuid"
} => my::fancy::Uuid
```

Not sure of the syntax. (We'll also want a way to specify the kinds of traits the given type impls.)

**Comment by ahl** [comment]
closed by #155
